tests/cross/src/test/scala/tests/pc/CompletionOverrideAllSuite.scala

```diff
@@ -0,0 +1,361 @@
+       |  val x = new Foo {
+       |    def@@
+       |  }
+       |}
+       |""".stripMargin,
+    "",
+    filter = _.contains("Implement")
+  )
+
+  check(
+    "same-scope",
+    """|package a { class Foo }
+       |package b { class Foo }
+       |
+       |trait Foo {
+       |  def one: a.Foo
+       |  def two: b.Foo
+       |}
+       |object Main {
+       |  val x = new Foo {
+       |    def@@
+       |  }
+       |}
+       |""".stripMargin,
+    """|Implement all members (2 total)
+       |""".stripMargin,
+    filter = _.contains("Implement")
+  )
+
+  checkEdit(
+    "same-scope-edit",
+    """|package a { class Foo }
+       |package b { class Foo }
+       |
+       |trait Foo {
+       |  def one: a.Foo
+       |  def two: b.Foo
+       |}
+       |object Main {
+       |  val x = new Foo {
+       |    def@@
+       |  }
+       |}
+       |""".stripMargin,
+    """|package a { class Foo }
+       |package b { class Foo }
+       |
+       |trait Foo {
+       |  def one: a.Foo
+       |  def two: b.Foo
+       |}
+       |object Main {
+       |  val x = new Foo {
+       |    def one: a.Foo = ${0:???}
+       |    def two: b.Foo = ${0:???}
+       |  }
+       |}
+       |""".stripMargin,
+    filter = _.contains("Implement")
+  )
+
+  check(
+    "include-val-var",
+    """|package example
+       |
+       |trait Foo {
+       |  def foo: Int
+       |  val bar: Int
+       |  var car: Int
+       |}
+       |object Main {
+       |  val x = new Foo {
+       |    def@@
+       |  }
+       |}
+       |""".stripMargin,
+    """|Implement all members (3 total)
+       |""".stripMargin,
+    filter = _.contains("Implement")
+  )
+
+  checkEdit(
+    "include-val-var-edit",
+    """|package example
+       |
+       |trait Foo {
+       |  def foo: Int
+       |  val bar: Int
+       |  var car: Int
+       |}
+       |object Main {
+       |  val x = new Foo {
+       |    def@@
+       |  }
+       |}
+       |""".stripMargin,
+    """|package example
+       |
+       |trait Foo {
+       |  def foo: Int
+       |  val bar: Int
+       |  var car: Int
+       |}
+       |object Main {
+       |  val x = new Foo {
+       |    def foo: Int = ${0:???}
+       |    val bar: Int = ${0:???}
+       |    var car: Int = ${0:???}
+       |  }
+       |}
+       |""".stripMargin,
+    filter = _.contains("Implement")
+  )
+
+  check(
+    "mixed-partial",
+    """|package example
+       |
+       |trait Foo {
+       |  def foo: Int
+       |  def foo2: Int
+       |  var bar: Int
+       |  var bar2: Int
+       |  val car: Int
+       |  val car2: Int
+       |}
+       |object Main {
+       |  val x = new Foo {
+       |   def foo: int = 3
+       |   var bar: int = 2
+       |   val car: Int = 1
+       |   def@@
+       |  }
+       |}
+       |""".stripMargin,
+    """|Implement all members (3 total)
+       |""".stripMargin,
+    filter = _.contains("Implement")
+  )
+
+  checkEdit(
+    "mixed-partial-edit",
```

Comment:
What happens if a method we override needs an import?
```scala
trait Foo {
  def file: java.io.File 
}
class Bar {
  def file@@ // implement all members
}
```

Reply:
Just tested it locally and it import it, but let me add a test to actually check it.

Reply:
Actually nevermind, it does for a single member but not like I thought it would with the implement all. Good catch, thanks.